SU2_CFD/src/solvers/CNSSolver.cpp

```diff
@@ -132,10 +140,15 @@ void CNSSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container, C
     SetPrimitive_Gradient_LS(geometry, config);
   }
 
-  /*--- Compute the limiter in case we need it in the turbulence model or to limit the
-   *    viscous terms (check this logic with JST and 2nd order turbulence model) ---*/
```

Comment:
I tried to do what this comment had been asking (for quite some time), because I think it was possible to make the solver do unnecessary work e.g. compute limiters and reconstruction gradients for centered schemes.
Also there was some mixing between flow and continuous adjoint options, which looking at the adjoint solvers code seemed redundant (it never uses the flow solver limiter for example). The regressions seem to agree.

I tested the main changes of this PR on some 3D problems and all seems fine, so I'll merge this soon for the next release.